### PR TITLE
FYST-1331-md-withdrawal-date-fix

### PR DESCRIPTION
**Pull request by DrewProebstel**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/FYST-1425
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

## What was done?
- Changes were made so that Maryland could actually withdraw funds after the 15th and before the 30th of april
## How to test?
- Use minimal in Maryland.
- change agi to 40k
- try different dates in taxes owed page

## Screenshots (for visual ch
Screenshot 2024-12-10 at 12 14 08 PM
anges)
Screenshot 2024-12-10 at 12 15 50 PM

Dangers:
The new way of determining withdrawal dates broke the Nc taxes owed tests. As is once a year the tests would need to be updated to new dates for the calendar year. Some code could be written to automatically determine which dates the tests need i.e  `def presidents_day(year)`  but I thought that could wait til after launch.

### CODE REVIEW

File: spec/forms/state_file/nc_taxes_owed_form_spec.rb [hunk 1]
```diff
@@ -87,7 +90,7 @@
       end
 
       context "when it's 5pm EST and withdrawal date is less than 2 business days from today" do
-        let(:app_time) { DateTime.parse("March 8th, #{current_year} 7pm EST").to_s } # Friday
+        let(:app_time) { DateTime.parse("March 10th, #{current_year} 7pm EST").to_s } # Friday
         let(:day) { "11" } # Monday, March 11th, 2024
```
Comment (arinchoi03) [inline spec/forms/state_file/nc_taxes_owed_form_spec.rb:94]: should we update this `day` to be actually Monday too?

File: spec/forms/state_file/nc_taxes_owed_form_spec.rb [hunk 2]
```diff
@@ -87,7 +90,7 @@
       end
 
       context "when it's 5pm EST and withdrawal date is less than 2 business days from today" do
-        let(:app_time) { DateTime.parse("March 8th, #{current_year} 7pm EST").to_s } # Friday
+        let(:app_time) { DateTime.parse("March 10th, #{current_year} 7pm EST").to_s } # Friday
```
Comment (arinchoi03) [inline spec/forms/state_file/nc_taxes_owed_form_spec.rb:93]: is this a Friday? This is a Monday for 2025

File: config/locales/es.yml [hunk 3]
```diff
@@ -278,7 +278,7 @@ es:
         is_hsa: es HSA
       taxes_owed:
         withdraw_amount_higher_than_owed: Ingresa una cantidad menor o igual a %{owed_amount}
-        withdrawal_date_deadline: Ingrese una fecha entre hoy y el 15 de abril de %{year} o antes
+        withdrawal_date_deadline: Por favor, ingrese una fecha entre hoy y hasta el %{day} de abril de %{year} inclusive
```
Comment (arinchoi03) [inline config/locales/es.yml:281]: are we flagging this translation for design folks to update?

File: spec/forms/state_file/taxes_owed_form_spec.rb [hunk 4]
```diff
@@ -86,6 +91,40 @@
             expect(intake.date_electronic_withdrawal).to eq Date.parse("April 15th, #{current_year}")
           end
         end
+
+        context "after ID's deadline and before MD's for MD intake" do
```
Comment (arinchoi03) [inline spec/forms/state_file/taxes_owed_form_spec.rb:95]: "after other states' deadline" rather than specifically ID I think

**Review by arinchoi03 — APPROVED** [review]
I have some q's on the test set up not lining up with the days, can tuple about it if it's easier to explain that way. The code implementation looks fine to me

File: spec/forms/state_file/nc_taxes_owed_form_spec.rb [hunk 5]
```diff
@@ -12,9 +12,12 @@
       payment_or_deposit_type: "mail"
     }
   end
-  let(:current_year) { "2024" }
+  let(:current_year) { (MultiTenantService.new(:statefile).current_tax_year + 1).to_s }
```
Comment (embarnard) [inline spec/forms/state_file/nc_taxes_owed_form_spec.rb:15]: I don't think we should change these dates. If we do we will have to change the test cases every year when current year changes. If this test is failing with updating those config dates we should find some other way to stub those dates in this test

Comment (DrewProebstel) [reply spec/forms/state_file/nc_taxes_owed_form_spec.rb:15]: You right im about to push up something better

File: spec/forms/state_file/nc_taxes_owed_form_spec.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@
 RSpec.describe StateFile::NcTaxesOwedForm do
   let!(:withdraw_amount) { 68 }
   let!(:intake) {
-    create :state_file_nc_intake,
+    create :state_file_ny_intake,
```
Comment (embarnard) [inline spec/forms/state_file/nc_taxes_owed_form_spec.rb:6]: oh oops this was my bad

File: spec/forms/state_file/taxes_owed_form_spec.rb [hunk 7]
```diff
@@ -86,6 +91,40 @@
             expect(intake.date_electronic_withdrawal).to eq Date.parse("April 15th, #{current_year}")
           end
         end
+
+        context "after other states's  deadline and before MD's for MD intake" do
```
Comment (arinchoi03) [inline spec/forms/state_file/taxes_owed_form_spec.rb:95]: i swear i thought `states'` is the right way to do this, but I'm not a native speaker lol

File: config/locales/es.yml (hunk 3 above)
Comment (arinchoi03) [reply config/locales/es.yml:281]: bump - not sure what the procedure for this is, but worth leaving a comment in jira story & tagging anu/madison

**Review by arinchoi03 — APPROVED** [review]

**Pull request merged**